tests/checkFrame.c

```diff
@@ -0,0 +1,269 @@
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+    You can contact the author at :
+    - LZ4 homepage : http://www.lz4.org
+    - LZ4 source repository : https://github.com/lz4/lz4
+*/
+
+/*-************************************
+*  Compiler specific
+**************************************/
+#ifdef _MSC_VER    /* Visual Studio */
+#  pragma warning(disable : 4127)        /* disable: C4127: conditional expression is constant */
+#  pragma warning(disable : 4146)        /* disable: C4146: minus unsigned expression */
+#endif
+
+
+/*-************************************
+*  Includes
+**************************************/
+#include "util.h"       /* U32 */
+#include <stdlib.h>     /* malloc, free */
+#include <stdio.h>      /* fprintf */
+#include <string.h>     /* strcmp */
+#include <time.h>       /* clock_t, clock(), CLOCKS_PER_SEC */
+#include <assert.h>
+#include "lz4frame.h"   /* include multiple times to test correctness/safety */
+#include "lz4frame.h"
+#define LZ4F_STATIC_LINKING_ONLY
+#include "lz4frame.h"
+#include "lz4frame.h"
+#include "lz4.h"        /* LZ4_VERSION_STRING */
+#define XXH_STATIC_LINKING_ONLY
+#include "xxhash.h"     /* XXH64 */
+
+
+/*-************************************
+*  Constants
+**************************************/
+#define KB *(1U<<10)
+#define MB *(1U<<20)
+#define GB *(1U<<30)
+
+
+/*-************************************
+*  Macros
+**************************************/
+#define DISPLAY(...)          fprintf(stderr, __VA_ARGS__)
+#define DISPLAYLEVEL(l, ...)  if (displayLevel>=l) { DISPLAY(__VA_ARGS__); }
+
+/**************************************
+*  Exceptions
+***************************************/
+#ifndef DEBUG
+#  define DEBUG 0
+#endif
+#define DEBUGOUTPUT(...) if (DEBUG) DISPLAY(__VA_ARGS__);
+#define EXM_THROW(error, ...)                                             \
+{                                                                         \
+    DEBUGOUTPUT("Error defined at %s, line %i : \n", __FILE__, __LINE__); \
+    DISPLAYLEVEL(1, "Error %i : ", error);                                \
+    DISPLAYLEVEL(1, __VA_ARGS__);                                         \
+    DISPLAYLEVEL(1, " \n");                                               \
+    exit(error);                                                          \
+}
+
+
+
+/*-***************************************
+*  Local Parameters
+*****************************************/
+static U32 no_prompt = 0;
+static U32 displayLevel = 2;
+static U32 use_pause = 0;
+
+
+/*-*******************************************************
+*  Fuzzer functions
+*********************************************************/
+#define MIN(a,b)  ( (a) < (b) ? (a) : (b) )
+#define MAX(a,b)  ( (a) > (b) ? (a) : (b) )
+
+void frameCheck(const char *srcFileName, unsigned bsid, size_t blockSize)
+{
+    FILE *srcFile;
+    LZ4F_decompressionContext_t dctx = NULL;
+    size_t srcBufferSize = 4 MB;
+    void *srcBuffer = malloc(srcBufferSize);
```

Comment:
coding style :
We like to `const` anytime possible.
This is a useful signal for reviewer / next programmer, who can then concentrate its attention to mutable (nont-`const`) variables.

In this case, buffers could be `void* const`, indicating that they are not going to be re-assigned later in the function.
Same for `srcBufferSize`, which never changes later in the function.

Unfortunately, we haven't yet found a tool to automate such check. So it's entirely reviewer - reliant.

Reply:
I added const for the buffer sizes, but it doesn't work for the pointers: they're passed to fread() and LZ4F_decompress() where they're not declared as const.

Reply:
> it doesn't work for the pointers:

A difficult point : don't confuse `void* const`, which is a fixed pointer to mutable data,
with `const void*`, which is a mutable pointer to read-only data.
The third variable `void const *` is actually the same as `const void*`.

`const void*` is incorrect, indeed : `fread()` will need to write into the buffer, so it's content must be mutable.

But `void* const` should be correct. I don't see any part of the code trying to modify the pointer itself.